### DISCUSSION TITLE
Crashing Error with Custom Character

### DISCUSSION THREAD

**Discussion by caserreal** [KadeArchive/Kade-Engine#517] [Q&A] [answered]
Okay so, I started making a mod like a month ago, tried to get help on how to fix my issue, and then took a break for a month. I'm trying to just test out my custom character, but it wont work. I know its something between the XML, Character.hx, or HealthIcon.hx . Everytime the game crashes, It doesn't give me a reason or a crash error. I'll send images of all 3 files i've listed below. Also, for the Char.hx i've copy + pasted the dad char.hx script and edited it.
![image](https://user-images.githubusercontent.com/82672666/119014101-a2da8f00-b965-11eb-9309-90c11bc6cc2a.png) 
![image](https://user-images.githubusercontent.com/82672666/119014264-cb628900-b965-11eb-845d-b6875c168b3f.png)
![image](https://user-images.githubusercontent.com/82672666/119023468-f05bf980-b96f-11eb-8652-b0dbfc302291.png)

**Comment by puyoxyz** [comment]
are you running with `lime run windows -debug`? the `-debug` option should give more info when it crashes i THINK

(replace `windows` with your platform (`mac`, `windows`, `linux`, `html5`))

**Reply by caserreal** [reply]
Okay, I'll try running it in the debug mode, thanks

**Reply by caserreal** [reply]
> are you running with `lime run windows -debug`? the `-debug` option should give more info when it crashes i THINK
> 
> (replace `windows` with your platform (`mac`, `windows`, `linux`, `html5`))

ok so, switching to my character in debug mode still crashed it, but the error message is "Null Object Reference"

**Comment by pepperly-art** [comment, marked as answer]
Try removing all the leading 1s in the animation frame numbers in your xml. So from  DOWN10000 to  DOWN0000, for example.

Third Down note should be "jan NOTE DOWN0002" and not "jan DOWN DOWN10002", too.

**Reply by caserreal** [reply]
Okay thanks, I'll try that and see if it works!

**Reply by caserreal** [reply]
> Try removing all the leading 1s in the animation frame numbers in your xml. So from DOWN10000 to DOWN0000, for example.
> 
> Third Down note should be "jan NOTE DOWN0002" and not "jan DOWN DOWN10002", too.

Sorry for late response but, It worked! Thanks so much. Only 1 problem: The character is very small in game, and is high in the air. I know i Can fix the X and Y location of the sprite with Offsets, but i'm not so sure about how to make the sprite bigger.

**Reply by isakube** [reply]
> but i'm not so sure about how to make the sprite bigger.

You could probably set a character specific scale in Character.hx. when a character is created, check and see if the character is "jan" and then tweak the scale to fit. i think

**Reply by caserreal** [reply]
Sorry not exactly sure what you mean, could you try to elaborate a little more?

**Reply by pepperly-art** [reply]
I blew girlfriend up by doing this; it might work for you. The sprite quality in comparison might be questionable, though. Drawing your character larger would be better.
`this.scale.set(4,4);`

![image](https://user-images.githubusercontent.com/82689031/119403784-e1957f80-bcac-11eb-8166-b3965234aee7.png)

![image](https://user-images.githubusercontent.com/82689031/119403808-eb1ee780-bcac-11eb-91c6-81b12721eb03.png)

**Reply by caserreal** [reply]
I'll try this and see how it works

**Reply by caserreal** [reply]
so uhh, I added the offsets, but the camera is really up above the sprite. Any suggestions on what to do?

**Reply by pepperly-art** [reply]
If you added the offsets to the xml directly, add a couple hundred to the frameHeight in the xml too, but if you added the offsets through the character's case in playstate.hx add too, I think:

`camPos.y += 300;`

You'll probably need to adjust your number (higher numbers = lower camera) until you're satisfied.

**Reply by caserreal** [reply]
I've been trying to edit the camera, hasnt been working. Also how can I edit the offsets in the playstate.hx ?

**Reply by pepperly-art** [reply]
Dang, the position did what I wanted to on a rough chart that didn't move the camera back to my character, and now that I'm retesting the code with a "proper" chart, it only changes the starting camera position and not the character's target camera position, for some reason. Sorry about that.

Looks like this section is where you want to target:
![image](https://user-images.githubusercontent.com/82689031/119711608-b7b79680-be2d-11eb-8d55-9abc857700f7.png)
-1000 makes the camera go way above my character:
![image](https://user-images.githubusercontent.com/82689031/119711445-922a8d00-be2d-11eb-8a5b-eb8b8dbe0ff7.png)

![image](https://user-images.githubusercontent.com/82689031/119711939-141ab600-be2e-11eb-904b-25377f98f5d4.png)
+1000 moves it way below, 
![image](https://user-images.githubusercontent.com/82689031/119711915-0cf3a800-be2e-11eb-940a-1dbf450fc220.png)

something like
```
case 'jan':
camFollow.y = dad.getMidpoint().y + 250;
```
but you'll have to figure out the best number for yourself.

**Reply by Megaman9009** [reply]
To change character size 				setGraphicSize(Std.int(width * number));
				                                       updateHitbox();

**Reply by Megaman9009** [reply]
I just copied it from senpais code. I suggest using decimals over 1 because 2 is SUPER BIG
